Link failure due to unused argument -pthread when using Clang and Makefiles

The error is as follows:

[ 73%] Building CXX object httpservice/CMakeFiles/usHttpService.dir/src/usServletConfig.cpp.o
cd /sandbox/fnannizz/CppMicroServices/build/httpservice && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++   -DUS_BUNDLE_NAME=usHttpService -DusHttpService_EXPORTS -I/sandbox/fnannizz/CppMicroServices/CppMicroServices/third_party -I/sandbox/fnannizz/CppMicroServices/build/include -I/sandbox/fnannizz/CppMicroServices/CppMicroServices/core/include -I/sandbox/fnannizz/CppMicroServices/build/core/include -I/sandbox/fnannizz/CppMicroServices/CppMicroServices/core/src/util -I/sandbox/fnannizz/CppMicroServices/CppMicroServices/core/src/service -I/sandbox/fnannizz/CppMicroServices/CppMicroServices/core/src/bundle -I/sandbox/fnannizz/CppMicroServices/CppMicroServices/httpservice/include -I/sandbox/fnannizz/CppMicroServices/build/httpservice/include -I/sandbox/fnannizz/CppMicroServices/CppMicroServices/httpservice/src  -std=c++11 -Werror -Wall -Wextra -Wpointer-arith -Winvalid-pch -Wcast-align -Wwrite-strings -Woverloaded-virtual -Wnon-virtual-dtor -Wold-style-cast -Wsign-promo -fdiagnostics-show-option -fstack-protector-all -Wno-mismatched-tags  -fvisibility-ms-compat -pthread -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -fPIC   -o CMakeFiles/usHttpService.dir/src/usServletConfig.cpp.o -c /sandbox/fnannizz/CppMicroServices/CppMicroServices/httpservice/src/usServletConfig.cpp
[ 74%] Linking CXX shared library ../lib/libusHttpService.dylib
cd /sandbox/fnannizz/CppMicroServices/build/httpservice && /usr/local/Cellar/cmake/3.6.0_1/bin/cmake -E cmake_link_script CMakeFiles/usHttpService.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++    -std=c++11 -Werror -Wall -Wextra -Wpointer-arith -Winvalid-pch -Wcast-align -Wwrite-strings -Woverloaded-virtual -Wnon-virtual-dtor -Wold-style-cast -Wsign-promo -fdiagnostics-show-option -fstack-protector-all -Wno-mismatched-tags  -fvisibility-ms-compat -pthread -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -dynamiclib -Wl,-headerpad_max_install_names    -compatibility_version 0.99.0 -o ../lib/libusHttpService.0.99.0.dylib -install_name /sandbox/fnannizz/CppMicroServices/build/lib/libusHttpService.0.99.0.dylib CMakeFiles/usHttpService.dir/us_init.cpp.o CMakeFiles/usHttpService.dir/__/third_party/civetweb/civetweb.c.o CMakeFiles/usHttpService.dir/__/third_party/civetweb/CivetServer.cpp.o CMakeFiles/usHttpService.dir/src/usHttpConstants.cpp.o CMakeFiles/usHttpService.dir/src/usHttpServlet.cpp.o CMakeFiles/usHttpService.dir/src/usServletContainer.cpp.o CMakeFiles/usHttpService.dir/src/usServletContext.cpp.o CMakeFiles/usHttpService.dir/src/usHttpOutputStreamBuffer.cpp.o CMakeFiles/usHttpService.dir/src/usHttpServletRequest.cpp.o CMakeFiles/usHttpService.dir/src/usHttpServletResponse.cpp.o CMakeFiles/usHttpService.dir/src/usServletConfig.cpp.o us_usHttpService/us_resources.o ../lib/libCppMicroServices.2.99.0.dylib -ldl
clang: error: argument unused during compilation: '-pthread' [-Werror,-Wunused-command-line-argument]
make[2]: *** [lib/libusHttpService.0.99.0.dylib] Error 1
make[1]: *** [httpservice/CMakeFiles/usHttpService.dir/all] Error 2
make: *** [all] Error 2

Clang bug report documenting that -pthread is not a supported linking option:
https://llvm.org/bugs/show_bug.cgi?id=7798

The fix will involve specifying -pthread as a build-only option when building with Clang.

--- a/cmake/usFunctionEmbedResources.cmake
+++ b/cmake/usFunctionEmbedResources.cmake
@@ -137,7 +137,7 @@ function(usFunctionEmbedResources)
     if(APPLE)
       add_custom_command(
         OUTPUT ${_source_output}
-        COMMAND ${CMAKE_CXX_COMPILER} -c ${US_CMAKE_RESOURCE_DEPENDENCIES_CPP} -o stub.o
+        COMMAND ${CMAKE_CXX_COMPILER} -isysroot ${CMAKE_OSX_SYSROOT} -c ${US_CMAKE_RESOURCE_DEPENDENCIES_CPP} -o stub.o
         COMMAND ${CMAKE_LINKER} -r -sectcreate __TEXT us_resources ${_zip_archive_name} stub.o -o ${_source_output}
         DEPENDS ${_zip_archive}
         WORKING_DIRECTORY ${_zip_archive_path}

